data_structures/stacks/postfix_evaluation.py

```diff
@@ -17,52 +24,197 @@
+                stack.append(x)  # append x to stack
+                if verbose:
+                    # output in tabular format
+                    print(
+                        str(x).rjust(8),
+                        ("push(" + str(x) + ")").ljust(12),
+                        stack,
+                        sep=" | ",
+                    )
+            elif is_operator(x):
+                # If only 1 value is inside stack and + or - is encountered
+                # then this is unary + or - case
+                if x in ["-", "+"] and len(stack) < 2:
+                    b = stack.pop()  # pop stack
+                    if x == "-":
+                        stack.append(-b)  # negate b and push again into stack
+                    else:  # when x is unary +
+                        stack.append(b)
+                    if verbose:
+                        # output in tabular format
+                        print(
+                            "".rjust(8),
+                            ("pop(" + str(b) + ")").ljust(12),
+                            stack,
+                            sep=" | ",
+                        )
+                        print(
+                            str(x).rjust(8),
+                            ("push(" + str(x) + str(b) + ")").ljust(12),
+                            stack,
+                            sep=" | ",
+                        )
+                else:
+                    b = stack.pop()  # pop stack
+                    if verbose:
+                        # output in tabular format
+                        print(
+                            "".rjust(8),
+                            ("pop(" + str(b) + ")").ljust(12),
+                            stack,
+                            sep=" | ",
+                        )
+
+                    a = stack.pop()  # pop stack
+                    if verbose:
+                        # output in tabular format
+                        print(
+                            "".rjust(8),
+                            ("pop(" + str(a) + ")").ljust(12),
+                            stack,
+                            sep=" | ",
+                        )
+                    # evaluate the 2 values popped from stack & push result to stack
+                    stack.append(opr[x](a, b))
+                    if verbose:
+                        # output in tabular format
+                        print(
+                            str(x).rjust(8),
+                            ("push(" + str(a) + str(x) + str(b) + ")").ljust(12),
+                            stack,
+                            sep=" | ",
+                        )
+            else:
+                print(f"{x} is neither a number, nor a valid operator")
+                break
+        # If everything executed correctly,the stack will contain
+        # only one element which is the result
+        result = stack[0] if len(stack) == 1 else None
+        return result
```

Comment:
If `len(stack) != 1` by the end of the computation, then it means that the program couldn't process the whole string, likely because it's not a valid postfix string. Thus, it should be fine to just raise an error:
```suggestion
        if len(stack) != 1:
            raise ArithmeticError("Input is not a valid postfix expression")
        return stack[0]
```